src/indieweb_utils/replies/context.py

```diff
@@ -7,320 +9,302 @@
-                    if h_entry["properties"].get("author") and h_entry["properties"]["author"][0].startswith("/"):
-                        author_url = url.split("/")[0] + "//" + domain + h_entry["properties"].get("author")[0]
-
-                    author = mf2py.parse(requests.get(author_url, timeout=10, verify=False).text)
-
-                    if author["items"] and author["items"][0]["type"] == ["h-card"]:
-                        author_url = h_entry["properties"]["author"][0]
-
-                        if author["items"][0]["properties"].get("name"):
-                            author_name = h_entry["properties"]["author"][0]["properties"]["name"][0]
-                        else:
-                            author_name = None
-
-                        if author["items"]["properties"].get("photo"):
-                            author_image = h_entry["properties"]["author"][0]["properties"]["photo"][0]
-                        else:
-                            author_image = None
-
-                if author_url is not None and author_url.startswith("/"):
-                    author_url = url.split("/")[0] + "//" + domain + author_url
-
-                if author_image is not None and author_image.startswith("/"):
-                    author_image = url.split("/")[0] + "//" + domain + author_image
-
-            if h_entry["properties"].get("content") and h_entry["properties"].get("content")[0].get("html"):
-                post_body = h_entry["properties"]["content"][0]["html"]
-                soup = BeautifulSoup(post_body, "html.parser")
-                post_body = soup.text
-
-                favicon = soup.find("link", rel="icon")
-
-                if favicon and not author_image:
-                    photo_url = favicon["href"]
-                    if not photo_url.startswith("https://") or not photo_url.startswith("http://"):
-                        author_image = "https://" + domain + photo_url
-                else:
-                    author_image = None
-
-                post_body = " ".join(post_body.split(" ")[:75]) + " ..."
-            elif h_entry["properties"].get("content"):
-                post_body = h_entry["properties"]["content"]
-
-                post_body = " ".join(post_body.split(" ")[:75]) + " ..."
+    photo: str
+    name: str
+    video: str
+    post_html: str
+    post_text: str
+    authors: List[PostAuthor]
+
+
+class ReplyContextRetrievalError(Exception):
+    def __init__(self, message):
+        self.message = message
+
+class UnsupportedScheme(Exception):
+    def __init__(self, message):
+        self.message = message
+
+def _generate_h_entry_reply_context(h_entry: dict, url: str, parsed_url: str, domain: str, webmention_endpoint_url: str) -> ReplyContext:
+    author_url = ""
+    author_name = ""
+    author_image = ""
+
+    if h_entry["properties"].get("author"):
+        if type(h_entry["properties"]["author"][0]) == dict and h_entry["properties"]["author"][0].get(
```

Comment:
It's better to use `isinstance` when checking the type of an object. e.g. 
```python
isinstance(h_entry["properties"]["author"][0], dict)
```

Reply:
Good suggestion!